src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,445 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class Duke {
+    private static final String TASKS_FILENAME = "./data/duke.txt";
+
+    private static final String INPUT_PROMPT = "$ ";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DELETE = "delete";
+
+    private static final String TASK_DEADLINE_SPLITTER = "/by";
+    private static final String TASK_EVENT_SPLITTER = "/at";
+
+    private static final String MESSAGE_WALL = "------------------------------------------------------------";
+    private static final String MESSAGE_LIST_EMPTY = "Task list is empty";
+    private static final String MESSAGE_LIST_HEADER = "Task List:";
+    private static final String MESSAGE_INVALID_TASK_NUMBER = "Invalid task number";
+    private static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
+    private static final String MESSAGE_WELCOME = "Hello! I'm Duke\nWhat can I do for you?";
+    private static final String MESSAGE_BYE = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_INVALID_TASK_TYPE = "Invalid task type";
+    private static final String MESSAGE_FILEWRITER_ERROR = "FileWriter Error";
+
```

Comment:
Good job grouping related constants together and separating them by spaces to improve code readability :)